cargo-annotation/annotation.sh

```diff
@@ -14,7 +14,7 @@ annotations=$(jq -r '
 ')
 
 # Print the annotations
-if [[ -n "$annotations" ]]; then
+if [[ -n "$annotations" && "${WITH_ANNOTATION}" == "true" ]]; then
```

Comment:
Shouldn't the `WITH_ANNOTATION == true` check be nested inside so that only the `echo $annotations` is guarded by it? We want to do `exit 1` regardless of whether GH annotations are enabled i.e failing the build is independent of whether we want GH annotations or not.